src/core/services/theming/theming.js

```diff
@@ -474,10 +504,49 @@ function ThemingProvider($mdColorPalette) {
       }
 
       /**
+       * Called when theme changes and md-watch-theme is set
+       * Changes theme of all themed children
+       */
+      function onWatchThemeChanged(newValue, oldValue)
```

Comment:
Does this function traverse down the dom tree to find mdTheme and apply the changes?

can't we use 'require' of directive and listen programmatically to changes??